File: package/main/src/IP/ipToLong.ts
/**
 * Converts an IPv4 address to a 32-bit number
 * @param {string} ip - IPv4 address to convert (e.g., "192.168.1.1")
 * @returns {number} 32-bit unsigned integer
 */
export const ipToLong = (ip: string): number => {
  const parts = ip.split(".");
  let result = 0;
  for (const part of parts) {
    result = ((result << 8) | Number.parseInt(part, 10)) >>> 0;
  }
  return result;
};

File: package/umt_python/src/ip/ip_to_long.py
def ip_to_long(ip: str) -> int:
    """
    Converts an IPv4 address to a 32-bit number.

    Args:
        ip: IPv4 address to convert (e.g., "192.168.1.1")

    Returns:
        32-bit unsigned integer

    Raises:
        ValueError: If IP address is invalid

    Example:
        >>> ip_to_long("192.168.1.1")
        3232235777
        >>> ip_to_long("0.0.0.0")
        0
    """
    if not ip:
        raise ValueError("IP address is required")

    parts = ip.split(".")
    if len(parts) != 4:
        raise ValueError("Invalid IP address format")

    result = 0
    for octet in parts:
        if (
            not octet.isascii()
            or not octet.isdigit()
            or (len(octet) > 1 and octet[0] == "0")
        ):
            raise ValueError("Invalid IP address format")

        num = int(octet)
        if num > 255:
            raise ValueError("Invalid IP address format")

        result = (result << 8) | num

    return result
